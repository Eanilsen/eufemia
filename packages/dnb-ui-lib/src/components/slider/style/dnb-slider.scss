/*
* DNB Slider
*
* ======================================================================== */

@import '../../../style/component/core-imports.scss';
// @import '../../../style/component/font-import.scss';

.dnb-slider {
  @include componentReset();
}

@import './_slider.scss';
