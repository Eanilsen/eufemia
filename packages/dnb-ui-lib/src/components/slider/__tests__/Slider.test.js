/**
 * Component Test
 *
 */

import React from 'react'
import {
  shallow,
  // mount,
  fakeProps,
  toJson,
  loadScss
} from '../../../core/jest/jestSetup'
import Component from '../Slider'
// just to make sure we re-run the test in watch mode due to changes in this file
import '../style/dnb-slider.scss'

const props = fakeProps(require.resolve('../Slider'), {
  optional: true
})

describe('Slider component', () => {
  // shallow compare the snapshot
  it('have to match snapshot', () => {
    const ComponentWrap = shallow(<Component {...props} href={null} />)
    expect(toJson(ComponentWrap)).toMatchSnapshot()
  })

  // then test the state management
  // const Comp = mount(<Component {...props} href={null} />)

  // it('has correct state after "change" trigger', () => {
  //   // default value value has to be false
  //   expect(Comp.state().value).toBe(false)
  //
  //   Comp.find('input').simulate('change') // we could send inn the event data structure like this: , { target: { value: true } }
  //   expect(Comp.state().value).toBe(true)
  //
  //   Comp.find('input').simulate('change')
  //   expect(Comp.state().value).toBe(false)
  //
  //   // also check if getDerivedStateFromProps sets the state as expected
  //   Comp.setProps({ value: true })
  //   expect(Comp.state().value).toBe(true)
  //
  //   const value = 'new value'
  //   Comp.setProps({ value })
  //   expect(Comp.find('input').props().value).toBe(value)
  // })
})

describe('Slider scss', () => {
  it('have to match snapshot', () => {
    const scss = loadScss(require.resolve('../style/dnb-slider.scss'))
    expect(scss).toMatchSnapshot()
  })
})
