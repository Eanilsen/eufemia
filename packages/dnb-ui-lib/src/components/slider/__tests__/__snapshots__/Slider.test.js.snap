// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider component have to match snapshot 1`] = `
<div
  className="dnb-slider className class slider__vertical"
>
  <div
    aria-orientation="vertical"
    aria-valuemax="max"
    aria-valuemin="min"
    aria-valuenow="value"
    className="slider__root slider__state--disabled"
    id="id"
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchMove={[Function]}
    onTouchStartCapture={[Function]}
    role="slider"
  >
    <div
      className="slider__line slider__line__before"
      style={
        Object {
          "height": "calc(NaN% - 6px)",
        }
      }
    />
    <button
      className="slider__thumb"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "top": "NaN%",
        }
      }
      tabIndex="0"
      type="button"
    />
    <div
      className="slider__line slider__line__after"
      style={
        Object {
          "height": "calc(NaN% - 6px)",
        }
      }
    />
  </div>
</div>
`;

exports[`Slider scss have to match snapshot 1`] = `
"/*
* DNB Slider
*
* ======================================================================== */
/*
 * Utilities
 * ========================================================================== */
/*
* Scopes
* ======================================================================== */
@font-face {
  font-family: 'Fedra Sans Std Book';
  src: url(\\"../../../../assets/fonts/FedraSansStd-Book.eot\\");
  src: url(\\"../../../../assets/fonts/FedraSansStd-Book.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts/FedraSansStd-Book.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts/FedraSansStd-Book.ttf\\") format(\\"truetype\\");
  font-weight: normal;
  font-style: normal; }

@font-face {
  font-family: 'Fedra Sans Std Demi';
  src: url(\\"../../../../assets/fonts/FedraSansStd-Demi.otf\\") format(\\"opentype\\"), url(\\"../../../../assets/fonts/FedraSansStd-Demi.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts/FedraSansStd-Demi.ttf\\") format(\\"truetype\\");
  font-weight: normal;
  font-style: normal; }

@font-face {
  font-family: 'Fedra Sans Std Medium';
  src: url(\\"../../../../assets/fonts/FedraSansStd-Medium.eot\\");
  src: url(\\"../../../../assets/fonts/FedraSansStd-Medium.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts/FedraSansStd-Medium.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts/FedraSansStd-Medium.ttf\\") format(\\"truetype\\");
  font-weight: 400;
  font-style: normal; }

.dnb-slider {
  font-family: var(--font-family-book);
  font-size: 1rem;
  font-weight: 300;
  font-style: normal;
  line-height: 1.5em; }
  .dnb-slider *,
  .dnb-slider ::before,
  .dnb-slider ::after {
    background-repeat: no-repeat;
    box-sizing: border-box; }
  .dnb-slider ::before,
  .dnb-slider ::after {
    text-decoration: inherit;
    vertical-align: inherit; }

/*
* Slider component
*
* ======================================================================== */
.dnb-slider {
  display: flex; }
  .dnb-slider .slider__root {
    width: 100%;
    cursor: pointer;
    padding: 1rem 0.5rem;
    position: relative; }
  .dnb-slider.slider__vertical {
    height: 100%;
    padding: 0.5rem 1rem; }
    .dnb-slider.slider__vertical .slider__root {
      width: inherit;
      height: inherit; }
  .dnb-slider.slider__reverse .slider__root {
    transform: scaleX(-1); }
  .dnb-slider.slider__reverse.slider__vertical .slider__root {
    transform: scaleY(-1); }
  .dnb-slider .slider__line {
    position: absolute;
    top: 50%;
    height: 2px;
    transform: translate(0, -50%);
    background-color: rgba(var(--color-sea-green), 0.5); }
  .dnb-slider .slider__state--disabled .slider__line {
    background-color: var(--color-black-30); }
  .dnb-slider.slider__vertical .slider__line {
    transform: translate(-50%, 0);
    left: 50%;
    top: initial;
    width: 2px; }
  .dnb-slider .slider__line__before {
    left: 0;
    z-index: 1; }
  .dnb-slider .slider__line__after {
    right: 0;
    opacity: 0.24; }
  .dnb-slider.slider__vertical .slider__line__before {
    top: 0; }
  .dnb-slider.slider__vertical .slider__line__after {
    bottom: 0; }
  .dnb-slider .slider__thumb {
    position: absolute;
    z-index: 2;
    height: 1.5rem;
    width: 1.5rem;
    transform: translate(-50%, -50%);
    margin: 0;
    padding: 0;
    display: inline-flex;
    align-items: center;
    vertical-align: middle;
    justify-content: center;
    cursor: pointer;
    user-select: none;
    outline: none;
    color: inherit;
    border: 1px solid var(--color-mint-green-50);
    border-radius: 50%;
    text-decoration: none;
    appearance: none;
    background-color: white;
    box-shadow: 1px 1px 2px rgba(0, 135, 137, 0.25), 0 0 1px rgba(0, 160, 163, 0.25); }
    .dnb-slider .slider__thumb:focus {
      outline: none;
      border-color: transparent; }
      html[data-whatinput='mouse'] .dnb-slider .slider__thumb:focus:not(.dnb-no-mouse-focus) {
        transition: box-shadow 0.1s ease-out 0s;
        box-shadow: 0 0 0 2px var(--color-emerald-green); }
      html[data-whatinput='keyboard'] .dnb-slider .slider__thumb:focus {
        box-shadow: 0 0 0 2px var(--color-signal-orange); }
  .dnb-slider .slider__state--disabled .slider__thumb {
    border: none;
    box-shadow: none;
    background-color: var(--color-black-30);
    height: 0.5rem;
    width: 0.5rem; }
  .dnb-slider .slider__state--jumped .slider__thumb,
  .dnb-slider .slider__state--jumped .slider__line {
    transition: width 250ms ease 0ms, height 250ms ease 0ms, left 250ms ease 0ms, top 250ms ease 0ms, box-shadow 250ms ease 0ms; }
  .dnb-slider .slider__state--activated .slider__thumb {
    border-width: 2px; }
  .dnb-slider .slider__state--disabled.slider__root,
  .dnb-slider .slider__state--disabled .slider__thumb {
    cursor: not-allowed; }
"
`;
